src/components/PivotTable/PivotTableGrid/util.ts

```diff
@@ -0,0 +1,579 @@
+    const gridArea = new GridArea(totalRowNumber, key, totalRowNumber + 1, key + 1)
+    divs.push({
+      types: new Set([PivotTableCellType.TOTAL, PivotTableCellType.VALUE]),
+      isEndRow: true,
+      gridArea: gridArea,
+      children: value,
+    })
+    cellPositions.add(gridArea.toString())
+  }
+}
+
+function buildMixedTableRowTotals(
+  mixedTableStartRowCache: Map<string, number>,
+  maxColumnEnd: number,
+  divs: PivotTableCellProps[],
+  cellPositions: Set<string>
+): (value: number, key: string, map: Map<string, number>) => void {
+  return (value, key) => {
+    const rowNumber = mixedTableStartRowCache.get(key)
+    if (rowNumber) {
+      const gridArea = new GridArea(rowNumber, maxColumnEnd + 1, rowNumber + 1, maxColumnEnd + 2)
+      divs.push({
+        types: new Set([PivotTableCellType.TOTAL, PivotTableCellType.VALUE]),
+        isEndColumn: true,
+        gridArea: gridArea,
+        children: value,
+      })
+      cellPositions.add(gridArea.toString())
+    }
+  }
+}
+
+function buildVerticalTableTotals<T extends object>(
+  isMixedTable: boolean,
+  maxRowEnd: number,
+  keys: (keyof T)[],
+  rowHeaderSpace: number,
+  maxColumnEnd: number,
+  divs: PivotTableCellProps[],
+  grandtotal: number,
+  cellPosition: Set<string>
+) {
+  let totalsGridArea: GridArea
+  if (isMixedTable) {
+    totalsGridArea = new GridArea(maxRowEnd, 1, maxRowEnd + 1, keys.length + 2)
+  } else {
+    totalsGridArea = new GridArea(rowHeaderSpace, maxColumnEnd, rowHeaderSpace + 1, maxColumnEnd + 1)
+    const totalGridArea = new GridArea(maxRowEnd, 1, maxRowEnd + 1, maxColumnEnd)
+    const dataValueGridArea = new GridArea(maxRowEnd, maxColumnEnd, maxRowEnd + 1, maxColumnEnd + 1)
+    divs.push({
+      types: new Set([PivotTableCellType.HEADER]),
+      isEndRow: true,
+      gridArea: totalGridArea,
+      children: TOTAL,
+    })
+    divs.push({
+      types: new Set([PivotTableCellType.TOTAL, PivotTableCellType.GRANDTOTAL, PivotTableCellType.VALUE]),
+      isEndRow: true,
+      isEndColumn: true,
+      gridArea: dataValueGridArea,
+      children: grandtotal,
+    })
+  }
+  divs.push({
+    types: new Set([PivotTableCellType.HEADER]),
+    isEndRow: isMixedTable,
+    isEndColumn: !isMixedTable,
+    gridArea: totalsGridArea,
+    children: TOTAL,
+  })
+  cellPosition.add(totalsGridArea.toString())
+}
+
+function getCurrentPath(filterKey: string, value: string): string {
+  return PATH_SEPARATOR + filterKey + '.' + value
+}
+
+/**
+ * Sums the span values of the parent cells to find the starting coordinates of the current cell.
+ */
+function getCellInitialPosition(initialPositionTree: CellInitialPosition | undefined) {
+  const stack: CellInitialPosition[] = []
+  let sum = 1
+
+  if (initialPositionTree) {
+    stack.push(initialPositionTree)
+  } else {
+    return 0
+  }
+
+  // Loop through the stack to calculate the sum of cellSpans
+  while (stack.length) {
+    const cell = stack.pop()
+    if (cell) {
```

Comment:
if you check at the start of the function if the initialPositionTree is undefined, you dont need to do this verification each iteration, right?

Reply:
Right